src/lib.rs

```diff
@@ -186,6 +189,55 @@ pub fn find_configuration_file() -> String {
     }
 }
 
+pub fn encrypt_single_file(
```

Comment:
Couple questions about this:

1. What was the reason behind not having this in the `encryption` module?
2. Looking at `encryption::encrypt_file` it looks like it's also encrypting a single file, it just takes an `EncryptionKey` instead. Maybe we can use a more specific name describing this, or could we maybe have a function that returns an `EncryptionKey` instead and we call the `encryption::encrypt_file` directly instead? I kind of like having a single `encrypt_file` function to avoid any synchronization issues between the two - for example, that extra `.expect("Unable to encrypt file");` which would probably be better if it's in the `bin.rs` instead as it's might be good to return the actual error from the library. What do you think?

Reply:
1. The main reason is that anything I consider "public" API I've been putting into `lib.rs` so that it can be easily adjusted work with C bridging. So if you think of `lib.rs` as the public interface for the "service layer", its functions are easy to call from `bin.rs`, or from C, and the rest of the API lives in appropriate places under there.   
2. Mostly the `lib.rs` file does conversion from strings into typed objects and back – mainly as a way to avoid having to repeat code between the C bridge and the binary. That said, all of the methods could be pushed down into the `encryption.rs` file, with shims for interacting with C. I'm not terribly set on it – would be happy to look at a PR with suggestions!!